### PR TITLE
Allow default_value for Optional arg

### PR DESCRIPTION
**Pull request by davidMcneil**
This allows setting a `default_value` for optional arguments. It is relevant to the discussion in #368.

Signed-off-by: David McNeil <mcneil.david2@gmail.com>

### CODE REVIEW

File: structopt-derive/src/attrs.rs [hunk 1]
```diff
@@ -310,6 +314,7 @@ impl Attrs {
                                 note = "see \
                                     https://docs.rs/structopt/0.3.5/structopt/#magical-methods")
                         };
+                        let ty = ty::sub_type(ty).unwrap_or(ty);
```
Comment (CreepySkeleton) [inline structopt-derive/src/attrs.rs:317]: Better to use `subtype_if_ty("Option")` here to keep things explicit

File: structopt-derive/src/attrs.rs [hunk 2]
```diff
@@ -7,7 +7,11 @@
 // except according to those terms.
 
 use crate::doc_comments::process_doc_comment;
-use crate::{parse::*, spanned::Sp, ty::Ty};
+use crate::{
+    parse::*,
+    spanned::Sp,
+    ty::{self, Ty},
```
Comment (CreepySkeleton) [inline structopt-derive/src/attrs.rs:13]: I'd import the functions explicitly rather than the module.

**Comment by davidMcneil** [comment]
After more testing, I realized this needed an addition to `clap` in order to correctly handle environment variables. See https://github.com/clap-rs/clap/pull/1826. For now, I changed `structopt`'s `clap` version to use the `clap` PR. When that PR is resolved, I will increment the `clap` version and remove the intermediate commits.

File: Cargo.toml [hunk 3]
```diff
@@ -27,7 +27,7 @@ paw = ["structopt-derive/paw"]
 travis-ci = { repository = "TeXitoi/structopt" }
 
 [dependencies]
-clap = { version = "2.33", default-features = false }
+clap = { git = "https://github.com/davidMcneil/clap.git", branch = "v2-master", default-features = false }
```
Comment (CreepySkeleton) [inline Cargo.toml:30]: Sorry for being blunt, but this is something we will definitely not accept. Only proper versions are our dependencies.

**Comment by CreepySkeleton** [comment]
> I will increment the clap version and remove the intermediate commits.

Oh well, this settles it. But to be fair, I doubt `clap` maintainers will be willing to unfreeze `v2-master` branch for this, sorry.

**Comment by davidMcneil** [comment]
> Oh well, this settles it. But to be fair, I doubt clap maintainers will be willing to unfreeze v2-master branch for this, sorry.

Ok, thanks. I will close this for now so it is not polluting the open PRs. I will look into adding it to clap v3. Rhanks for your feedback.

**Pull request closed without merging**